src/app/cli/src/commands/system_diagnose_command.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use container_runtime::{ContainerRuntime, ContainerRuntimeType};
+
+use super::{CLIError, Command};
+use crate::output::*;
+use crate::WorkspaceService;
+
+pub struct SystemDiagnoseCommand {
+    output_config: Arc<OutputConfig>,
+    output_format: Option<String>,
+}
+
+impl SystemDiagnoseCommand {
+    pub fn new<S>(output_config: Arc<OutputConfig>, output_format: Option<S>) -> Self
+    where
+        S: Into<String>,
+    {
+        Self {
+            output_config,
+            output_format: output_format.map(|s| s.into()),
+        }
+    }
+}
+
+#[async_trait::async_trait(?Send)]
+impl Command for SystemDiagnoseCommand {
+    fn needs_workspace(&self) -> bool {
+        false
+    }
+
+    async fn run(&mut self) -> Result<(), CLIError> {
+        write_output(
+            SystemDiagnose::check().await,
+            &self.output_config,
+            self.output_format.as_ref(),
+        )?;
+        Ok(())
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SystemDiagnose {
+    pub run_check: RunCheck,
+}
+
+impl SystemDiagnose {
+    pub async fn check() -> Self {
+        Self {
+            run_check: RunCheck::collect().await,
+        }
+    }
+}
+
+#[derive(Debug, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RunCheck {
+    pub container_runtime_type: ContainerRuntimeType,
+    pub is_installed: bool,
+    pub is_rootless: bool,
+    pub workspace_dir: Option<String>,
+    pub is_workspace_consitence: bool,
+}
+
+impl RunCheck {
+    pub async fn collect() -> Self {
+        let container_runtime = ContainerRuntime::default();
+        let is_container_command_err = container_runtime
+            .custom_cmd("--version".to_string())
+            .output()
+            .await
+            .is_err();
+
+        let workspace_layout = WorkspaceService::find_workspace();
+        let current_workspace = WorkspaceService::new(Arc::new(workspace_layout.clone()));
+
+        let is_rootless = match container_runtime
+            .custom_cmd(r"info".to_string())
+            .output()
+            .await
+        {
+            Ok(container_info) => String::from_utf8(container_info.stdout)
+                .unwrap()
+                .contains("rootless: true"),
```

Comment:
Should we also tests that we can spawn some container and that we can mount a file into it?

`rootless: true` is hardly an indication that podman can function correctly.